### PR TITLE
Add prometheus integration

### PR DESCRIPTION
**Pull request by arturo-seijas**

### CODE REVIEW

**Comment by merkata** [comment]
Looks really good, the PR is very clear in terms of changes. I like how container events on pebble collapse into a single handler function, seems like a good pattern :) 
Do you plan on having an integration test for prometheus, seeing it scrapes something etc, is that planned?
Good work!

**Comment by arturo-seijas** [comment]
> Looks really good, the PR is very clear in terms of changes. I like how container events on pebble collapse into a single handler function, seems like a good pattern :) Do you plan on having an integration test for prometheus, seeing it scrapes something etc, is that planned? Good work!

No, I don't plan to add an integration test for prometheus. The scraping falls under the prometheus charm responsibility hence, that'd be the proper place to add it.
Thank you for your kind words :)

**Comment by mthaddon** [comment]
Seems like the integration tests failed here - can you take a look?

**Comment by arturo-seijas** [comment]
> Seems like the integration tests failed here - can you take a look?

Fixed

File: files/etc/nginx/nginx.conf [hunk 1]
```diff
@@ -37,5 +37,9 @@ http {
       add_header 'Content-Type' 'application/json';
       return 204;
     }
+
+    location = /stub_status {
+      stub_status;
+    }
```
Comment (mthaddon) [inline files/etc/nginx/nginx.conf:43]: I wonder if we need to/can restrict this to only being accessible from 127.0.0.1? I don't think we'd want that URL to be accessible to end users would we?

Comment (arturo-seijas) [reply files/etc/nginx/nginx.conf:45]: Indeed. Good catch

**Review by mthaddon — APPROVED** [review]
Looks great, thanks!

**Pull request merged**